pkg/types/validators_test.go

```diff
@@ -0,0 +1,46 @@
+package types
+
+import (
+	"testing"
+
+	validator "github.com/go-playground/validator/v10"
+	"github.com/stretchr/testify/assert"
+)
+
+type TestGitProjectValidators struct {
+	t *testing.T
+}
+
+func NewTestGitProjectValidator(t *testing.T) *TestGitProjectValidators {
+	p := &TestGitProjectValidators{
+		t: t,
+	}
+	return p
+}
+
+func TestValidateGitProjectSubDir(t *testing.T) {
+
+	var subDirItems = []struct {
+		have string
+		want bool
+	}{
+		{"subDir", true},
+		{"subDir/../subDir", true},
+		{"subDir?", false},                        // wrong characters
+		{"subDir/../sub?D|ir", false},             // wrong characters
+		{"subDir/../../../subDir/subDir2", false}, // escape from git repo root
```

Comment:
No need to check for escaping as this is done in another place already: [deployment_project.go#L42](https://github.com/kluctl/kluctl/blob/d6aee5b1d63dcd75370c009b9023e862c589f1d5/pkg/deployment/deployment_project.go#L42)